ntoskrnl/ke/amd64/kiinit.c

```diff
@@ -429,15 +429,15 @@ KiSystemStartup(IN PLOADER_PARAMETER_BLOCK LoaderBlock)
     FrLdrDbgPrint = LoaderBlock->u.I386.CommonDataArea;
     //FrLdrDbgPrint("Hello from KiSystemStartup!!!\n");
 
-    /* Save the loader block */
-    KeLoaderBlock = LoaderBlock;
-
     /* Get the current CPU number */
     Cpu = KeNumberProcessors++; // FIXME
 
     /* LoaderBlock initialization for Cpu 0 */
     if (Cpu == 0)
     {
+        /* Save the loader block */
+        KeLoaderBlock = LoaderBlock;
```

Comment:
If this hasn't been merged yet for x86, it's in @DarkFire01's branch(es) and should be merged at some point soon. There is no need for an ASSERT, since it should work as well, if the loader block was different.